### ISSUE TITLE
Cannot use collumn selector in left join

### ISSUE THREAD

**Issue by bram96** [crate/crate#4341]
I tried to do the following query: https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/94514/3508063/MFg2uRl8DMgxhD7/-.txt

I get the following error: https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/94514/3508063/AtRHcgrZ1MN8hcK/-.txt

It is working well when I do select \* but I cannot select a single collumn
##

**Comment by mfussenegger** [comment]
Can confirm that this happens on 0.56.X and also on master.

We'll take a closer look at it. Thanks for reporting


```
drop table if exists t1;
drop table if exists t2;

create table t1 (id int, some_id int);
create table t2 (id int, t1_id int, other_id int);

insert into t1 (id) values (1);
insert into t2 (id, t1_id, other_id) values (1, 1, 1);


select 
    m.id 
from t1 m
    left join t2 c on c.t1_id = m.id
where 
    m.some_id = 1
    and c.other_id in (1)
limit 100;
```

**Comment by seut** [comment]
We've fixes this by https://github.com/crate/crate/commit/958ebf3d5f2c684564acb89fe9aa1bf82c58b808. Will be released with the next 0.57 & 0.56 hotfix releases. Thanks for reporting!
